### DISCUSSION TITLE
Consider importing the dnlib module

### DISCUSSION THREAD

**Discussion by glenn-slayden** [dnSpyEx/dnSpy#195] [General]
As far as I can tell, the [dnlib](https://github.com/0xd4d/dnlib) module is not currently forked into this repo. It was last updated on Nov 19, 2022, but I'm not sure if it's actively maintained, defunct, or ...? Unlike the situation with [dnSpy-*classic*™](https://github.com/dnSpy/dnSpy) repo, *dnlib* is not currently marked as being "archived."

I found at least one bug in *dnlib* that I am willing or eager to propagate into [dnSpyEx](https://github.com/dnSpyEx), but I'm not sure how to proceed with that, given the current state of affairs.

The bug I tracked down is a for a case where dnSpyEx crashes while attempting to read the PDB symbols format for a C++/CLI mixed-mode executable. The fault is within dnlib, which deliberately faults the operation when due to misinterpreting some size calculations involving poorly understood header field values.

Finally on this subject FWIW, I can report that importing the stable dnlib source into (er, my fork of) this repo was a completely trivial and effortless process.

**Comment by ElektroKill** [comment]
Hi, `dnlib` is from my knowledge currently in a community-maintained state. The author accepts PRs with non-breaking changes/additions/fixes and creates a NuGet release every so often. I suggest you try your luck with submitting a PR to dnlib with a fix for the bug you found. I think submitting pull requests to the original project is better so that other consumers can also benefit from them (I'm also not really a fan of Git submodules)
